### DISCUSSION TITLE
Retrieve Refresh_Token from Backend with Cookie from other Domain

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#9788] [Archive]
Yes I have Cookie: Secure and SameSite=None

For example: calling: https://se-services.informatik.uos.de/myapp/api/auth/login/studiptest?redirect=http://localhost?refresh_token
Will bring me to localhost but from there i can't obtain the token, since no call to the backend will add the cookie.

Can someone provide an javascript example on which he get's a refresh_token from my page?
I am trying this now since 2 weeks.

**Comment by NilsBaumgartner1994** [comment]
PS: You can fill any credentials you like to pass the SSO.

**Comment by TomS-** [comment]
I don't know if this is related but I'm having a similar issue:
https://github.com/directus/directus/discussions/9720

Trouble is, we had a Quiz App go live and the client isn't too happy because revisiting users get stuck logged in but can't do anything. This is because `auth_token` is set, but no `refresh_token` was actioned, so `directus.auth.token` is true but `directus.users.me` can't be read as it's expired and they can not logout because logging out requires a `refresh_token` set in the cookie. So they are stuck in limbo. I also have `Cookie: Secure` and `SameSite=None`.

**Reply by TomS-** [reply]
I've offered to let them remote into my computer for debugging in case it's a rare use-case. But no reply as of yet.

**Comment by joselcvarela** [comment]
Hello @NilsBaumgartner1994 
I have successfully set up this scenario with Google OpenID.

So this is on `.env` on Directus side :
```sh
AUTH_PROVIDERS="google"

AUTH_GOOGLE_DRIVER="openid"
AUTH_GOOGLE_CLIENT_ID="XXXX.googleusercontent.com"
AUTH_GOOGLE_CLIENT_SECRET= "XXXXXX"
AUTH_GOOGLE_ISSUER_URL="https://accounts.google.com"
AUTH_GOOGLE_IDENTIFIER_KEY="email"
AUTH_GOOGLE_ICON="google"
AUTH_GOOGLE_ALLOW_PUBLIC_REGISTRATION="true"
AUTH_GOOGLE_DEFAULT_ROLE_ID="XXXXXX"
```

And this is my client in React:
```tsx
import React, { useState } from "react";
import "./App.css";
import { useEffect } from "react";

import { Directus } from "@directus/sdk";

const directus = new Directus("http://localhost:8055");

function App() {
  const [me, set_me]: [any, (me: any) => void] = useState(null);

  useEffect(() => {
    async function start() {
      await directus.auth.refresh();
      set_me(await directus.users.me.read().catch(() => null));
    }

    start();
  }, []);

  return (
    <div className="App">
      {me ? <h1>Authenticated</h1> : <h1>NOT Authenticated</h1>}
      {me ? null : (
        <a href="http://localhost:8055/auth/login/google?redirect=http://127.0.0.1:3000">
          authenticate
        </a>
      )}
      {me ? <pre>{JSON.stringify(me)}</pre> : null}
    </div>
  );
}

export default App;

```
___
Edit:

This should also be on `.env`
```sh
REFRESH_TOKEN_COOKIE_DOMAIN="XXXX" # This should be the domain where Directus is running, i.e. the domain part of PUBLIC_URL
REFRESH_TOKEN_COOKIE_SECURE="true"
REFRESH_TOKEN_COOKIE_SAME_SITE="None"
```

**Comment by NilsBaumgartner1994** [comment]
You have still the same domain (localist) for your Frontend and directus


Beste Grüße,
Nils Baumgartner

**Reply by joselcvarela** [reply]
Hey again,
Yep, my bad. I have edited and change `?redirect=http://localhost:3000"` by this `redirect=http://127.0.0.1:3000"`
And experienced the not logged in flow. So I investigate and found that the following should be set on `.env`:
```sh
REFRESH_TOKEN_COOKIE_DOMAIN="XXXX" # This should be the domain where Directus is running, i.e. the domain part of PUBLIC_URL
REFRESH_TOKEN_COOKIE_SECURE="true"
REFRESH_TOKEN_COOKIE_SAME_SITE="None"
```

I have updated my previous answer.
